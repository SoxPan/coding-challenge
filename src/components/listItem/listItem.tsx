import React, { FC } from "react";
import styled from "styled-components";
import { BrowserRouter, Link } from "react-router-dom";

import { IJob } from "../../types";

interface IProps {
    job: IJob;
}

const StyledListItem = styled.li`
    width: 100%;
    background: #fff;
    border-bottom: 2px solid #e0e0e0;

    &:last-child {
        border-bottom: none;
    }

    &:hover {
        background: #e0e0e0;
    }
`;

export const StyledListitemLink = styled(Link)`
    width: 100%;
    height: auto;
    display: block;
    padding-left: 15px;
    text-decoration: none;
    line-height: 39px;
    color: #000;
`;

export const ListItem: FC<IProps> = ({ job }) => {
    return (
        <StyledListItem>
            <StyledListitemLink to={`/${job.id}`}>
                {job.title}
            </StyledListitemLink>
        </StyledListItem>
    );
};
