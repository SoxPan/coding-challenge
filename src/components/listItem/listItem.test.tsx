import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { StyledListitemLink } from "./listItem";
import { ListItem } from "./listItem";
import data from "../../../public/mock.json";

describe("<ListItem />", () => {
    it("Item of list renders correctly", () => {
        const tree = renderer.create(<ListItem job={data[0]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Item of list renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<ListItem job={data[0]} />, div);
    });

    it("Should change route correctly", () => {
        const wrapper = shallow(<ListItem job={data[0]} />);
        expect(wrapper.find(StyledListitemLink).length).toBe(1);
        expect(wrapper.find(StyledListitemLink).props().to).toBe("/1");
    });
});
