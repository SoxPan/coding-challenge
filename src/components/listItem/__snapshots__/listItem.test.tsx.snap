// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> Item of list renders correctly 1`] = `
<li
  className="sc-bdVaJa ekTLhB"
>
  <a
    className="sc-bwzfXH hCIMcE"
    href="/1"
    onClick={[Function]}
  >
    Frontend Developer
  </a>
</li>
`;
